src/QMCWaveFunctions/Jastrow/OneBodyJastrowOrbitalBsplineSoA.h

```diff
@@ -0,0 +1,193 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+    
+    
+#ifndef ONE_BODY_JASTROW_ORBITAL_BSPLINE_SOA_H
+#define ONE_BODY_JASTROW_ORBITAL_BSPLINE_SOA_H
+
+#include "Particle/DistanceTableData.h"
+#include "Particle/DistanceTable.h"
+#include "QMCWaveFunctions/Jastrow/J1OrbitalSoA.h"
+#include "QMCWaveFunctions/Jastrow/BsplineFunctor.h"
+#include "QMCWaveFunctions/Jastrow/CudaSpline.h"
+#include "QMCWaveFunctions/Jastrow/BsplineJastrowCuda.h"
+#include "QMCWaveFunctions/Jastrow/BsplineJastrowCudaPBC.h"
+#include "Configuration.h"
+
+namespace qmcplusplus
+{
+
+template<class FT>
+class OneBodyJastrowOrbitalBsplineSoA :
+  public J1OrbitalSoA<FT>
```

Comment:
We only have BsplineFunctor working on GPU. So there is no need to make it a template function. The old way is sufficient.